### PR TITLE
[Feature] Conference calling

### PR DESCRIPTION
**Pull request by johnxnguyen**
## What's new in this PR?

This is the feature branch for **Conference Calling**. All changes in this PR have already been reviewed in separate PRs, see the corresponding commits.

## Notes

This PR contains an initial yet functional implementation. All conference calling features are behind a feature flag with is currently disabled for all builds except the AVS build.

**JIRA:** 

https://wearezeta.atlassian.net/browse/ZIOS-12708
https://wearezeta.atlassian.net/browse/ZIOS-13454

### CODE REVIEW

File: Source/Calling/AVSWrapper+Handlers.swift [hunk 1]
```diff
@@ -19,53 +19,172 @@
...
+        typealias ContextRef = VoidPtr
+
+        /// Callback used to inform user that call uses CBR (in both directions).
+        ///
+        /// typedef void (wcall_audio_cbr_change_h)(const char *userid,
+        ///                                         const char *clientid,
+        ///                                         int enabled,
+        ///                                         void *arg);
+
+        typealias ConstantBitRateChange = @convention(c)(StringPtr, StringPtr, Int32, ContextRef) -> Void
```
Comment (David-Henner) [inline Source/Calling/AVSWrapper+Handlers.swift:37]: syntax: `@convention(c)(StringPtr,...` no space after `(c)` is inconsistent with the other declarations

File: Source/Calling/WireCallCenterV3+Events.swift [hunk 2]
```diff
@@ -79,10 +104,14 @@ extension WireCallCenterV3 {
     }
 
     /// Handles incoming calls.
-    func handleIncomingCall(conversationId: UUID, messageTime: Date, userId: UUID, isVideoCall: Bool, shouldRing: Bool) {
+    func handleIncomingCall(conversationId: UUID, messageTime: Date, client: AVSClient, isVideoCall: Bool, shouldRing: Bool) {
         handleEvent("incoming-call") {
-            let callState : CallState = .incoming(video: isVideoCall, shouldRing: shouldRing, degraded: self.isDegraded(conversationId: conversationId))
-            self.handleCallState(callState: callState, conversationId: conversationId, userId: userId, messageTime: messageTime)
+            let isDegraded = self.isDegraded(conversationId: conversationId)
```
Comment (David-Henner) [inline Source/Calling/WireCallCenterV3+Events.swift:109]: should `self` be made `weak` in the block?

File: Source/Calling/WireCallCenterV3+Events.swift [hunk 3]
```diff
@@ -96,22 +125,35 @@ extension WireCallCenterV3 {
...
+            }
         }
     }
 
     /// Handles established calls.
-    func handleEstablishedCall(conversationId: UUID, userId: UUID) {
+    func handleEstablishedCall(conversationId: UUID) {
         handleEvent("established-call") {
-            self.handleCallState(callState: .established, conversationId: conversationId, userId: userId)
+            // WORKAROUND: the call established handler will is called once for every participant in a
```
Comment (David-Henner) [inline Source/Calling/WireCallCenterV3+Events.swift:146]: typo: `handler will is called`

Comment (johnxnguyen) [reply Source/Calling/WireCallCenterV3+Events.swift:146]: Fixed.

**Comment by johnxnguyen** [comment]
@zenkins release major

**Pull request merged**